x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/transform/action/GetCheckpointAction.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.core.transform.action;
+
+import org.elasticsearch.action.ActionRequest;
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.action.ActionResponse;
+import org.elasticsearch.action.ActionType;
+import org.elasticsearch.action.IndicesRequest;
+import org.elasticsearch.action.support.IndicesOptions;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Objects;
+
+public class GetCheckpointAction extends ActionType<GetCheckpointAction.Response> {
+
+    public static final GetCheckpointAction INSTANCE = new GetCheckpointAction();
+
+    // note: this is an admin action, it must be called with user headers
+    public static final String NAME = "indices:internal/transform/checkpoint";
+
+    private GetCheckpointAction() {
+        super(NAME, GetCheckpointAction.Response::new);
+    }
+
+    public static class Request extends ActionRequest implements IndicesRequest.Replaceable {
```

Comment:
`IndicesRequest.Replaceable` has a default method `boolean allowsRemoteIndices()` that returns `false`, which means the `indices` field cannot contain remote index names of the format `cluster_alias:index`. My reading of the code seems to suggest that all indices here have their cluster_alias prefix stripped (if any). So it should be OK for the class to rely on the default implementation of `allowsRemoteIndices()`. But it would be great if you could confirm on this.

Reply:
yes, that's correct. CCS is handled before calling this action: the `source` (indices to query) is split per cluster, get checkpoint is called separately for every cluster. This has been implemented like this for the original checkpointing, which as explained above is kept for BWC. It would be quite complex to separate the 2. Once the old style checkpointing got removed somewhen in future, we might revisit.